fenix/app/src/androidTest/java/org/mozilla/fenix/ui/ComposeTabbedBrowsingTest.kt

```diff
@@ -0,0 +1,380 @@
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ *  Tests for verifying basic functionality of tabbed browsing
+ *
+ *  Including:
+ *  - Opening a tab
+ *  - Opening a private tab
+ *  - Verifying tab list
+ *  - Closing all tabs
+ *  - Close tab
+ *  - Swipe to close tab (temporarily disabled)
+ *  - Undo close tab
+ *  - Close private tabs persistent notification
+ *  - Empty tab tray state
+ *  - Tab tray details
+ *  - Shortcut context menu navigation
+ */
+
+class ComposeTabbedBrowsingTest {
+    private lateinit var mDevice: UiDevice
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule(order = 0)
+    val composeTestRule =
+        AndroidComposeTestRule(
+            HomeActivityTestRule.withDefaultSettingsOverrides(
+                tabsTrayRewriteEnabled = true,
+            ),
+        ) { it.activity }
+
+    @Rule(order = 1)
+    @JvmField
+    val retryTestRule = RetryTestRule(3)
+
+    @Before
+    fun setUp() {
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun openNewTabTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+            mDevice.waitForIdle()
+            verifyTabCounter("1")
+        }.openComposeTabDrawer(composeTestRule) {
+            verifyNormalBrowsingButtonIsSelected()
+            verifyExistingOpenTabs("Test_Page_1")
+            closeTab()
+        }.openTabDrawer {
+            verifyNoOpenTabsInNormalBrowsing()
+        }.openNewTab {
+        }.submitQuery(defaultWebPage.url.toString()) {
+            mDevice.waitForIdle()
+            verifyTabCounter("1")
+        }.openComposeTabDrawer(composeTestRule) {
+            verifyNormalBrowsingButtonIsSelected()
+            verifyExistingOpenTabs("Test_Page_1")
+        }
+    }
+
+    @Test
+    fun openNewPrivateTabTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        homeScreen {}.togglePrivateBrowsingMode()
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+            mDevice.waitForIdle()
+            verifyTabCounter("1")
+        }.openComposeTabDrawer(composeTestRule) {
+            verifyPrivateTabsList()
+            verifyPrivateBrowsingButtonIsSelected()
+        }.toggleToNormalTabs {
+            verifyNoOpenTabsInNormalBrowsing()
+        }.toggleToPrivateTabs {
+            verifyPrivateTabsList()
+        }
+    }
+
+    @Ignore("Being converted in: https://bugzilla.mozilla.org/show_bug.cgi?id=1831602")
```

Comment:
This matches the bug number of this patch. Should this test be included now?

Reply:
oh, that's a goof on my end. I'll fix this with the next push